diff --git a/debian/NEWS b/debian/NEWS
index dd09744..de7b640 100644
--- a/debian/NEWS
+++ b/debian/NEWS
@@ -1,3 +1,9 @@
+postgrey (1.35-1+miab1)
+
+  Added DNSWL.org whitelisting.
+
+ -- Joshua Tauberer <jt@occams.info>  Mon May 18 18:58:40 EDT 2015
+
 postgrey (1.32-1) unstable; urgency=low
 
   Postgrey is now listening to port 10023 and not 60000.  The latter was an
diff --git a/debian/changelog b/debian/changelog
index 1058e15..e5e3557 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+postgrey (1.35-1+miab1) trusty; urgency=low
+
+  * Added DNSWL.org whitelisting.
+
+ -- Joshua Tauberer <jt@occams.info>  Mon, 18 May 2015 21:58:40 +0000
+
 postgrey (1.35-1) unstable; urgency=low
 
   * New upstream release (Closes: 756486)
diff --git a/debian/control b/debian/control
index ce12ba6..0a82855 100644
--- a/debian/control
+++ b/debian/control
@@ -1,14 +1,11 @@
 Source: postgrey
 Section: mail
 Priority: optional
-Maintainer: Antonio Radici <antonio@debian.org>
-Uploaders: Jon Daley <jondaley-guest@alioth.debian.org>
+Maintainer: Joshua Tauberer <jt@occams.info>
 Build-Depends: debhelper (>= 7), quilt
 Build-Depends-Indep: po-debconf
 Standards-Version: 3.9.6
 Homepage: http://postgrey.schweikert.ch/
-Vcs-Browser: http://git.debian.org/?p=collab-maint/postgrey.git
-Vcs-Git: git://git.debian.org/git/collab-maint/postgrey.git
 
 Package: postgrey
 Architecture: all
@@ -25,3 +22,6 @@ Description: greylisting implementation for Postfix
  .
  While Postgrey is designed for use with Postfix, it can also be used
  with Exim.
+ .
+ This version has been modified by Mail-in-a-Box to whitelist senders
+ in the DNSWL.org list. See https://mailinabox.email.
diff --git a/debian/copyright b/debian/copyright
index 3cbe377..bf09b89 100644
--- a/debian/copyright
+++ b/debian/copyright
@@ -1,6 +1,10 @@
+This package is a fork by Mail-in-a-Box (https://mailinabox.email). Original
+copyright statement follows:
+----------------------------------------------------------------------------
+
 This Debian package was prepared by Adrian von Bidder <cmot@debian.org> in
 July 2004, then the package was adopted by Antonio Radici <antonio@dyne.org>
-in Sept 2009
+in Sept 2009.
 
 It was downloaded from http://postgrey.schweikert.ch/
 
diff --git a/debian/patches/series b/debian/patches/series
index f4c5e31..3cd62b8 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -1,3 +1,3 @@
 imported-upstream-diff
 disable-transaction-logic
-
+mailinabox
